platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/ProblemReportingCrossProjectModelAccess.kt

```diff
@@ -121,6 +119,10 @@ class ProblemReportingCrossProjectModelAccess(
         }
     }
 
+    override fun getAllprojectsForGradle(gradle: GradleInternal): MutableSet<out ProjectInternal> {
+        return getAllprojects(gradle.getRootProject(), gradle.getRootProject());
```

Comment:
❓ IIUC, this will *not* wrap the root project, as opposed to the root project being wrapped in `DefaultCrossProjectModelAccess` in the same method.

Would that mean there is going to be a difference in behavior when configuring the root project with IP and without IP?

Consider a case like this:
```
// settings

gradle.rootProject {
    // mutable access 
}

gradle.lifecycle.beforeProject {
    // do something
}
```

Without IP, this will execute as
```
gradle.rootProject
gradle.lifecycle.beforeProject (triggered by mutable access)
mutable access
```

With IP, it seems it will execute as
```
gradle.rootProject
mutable access
gradle.lifecycle.beforeProject 
```

Reply:
The same discrepancy seems to be true if instead of `gradle.rootProject { <mutable access> }` one would use `gradle.allprojects { if (<is root>) { <mutable access> } }`